core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java

```diff
@@ -112,11 +112,13 @@ private RetryHandlingBlockWorkerClient(
   private void init() throws IOException {
     if (mSessionId != null) {
       // Register the session before any RPCs for this session start.
-      try {
-        sessionHeartbeat();
-      } catch (InterruptedException e) {
-        throw Throwables.propagate(e);
-      }
+      retryRPC(new RpcCallable<Void, BlockWorkerClientService.Client>() {
+        @Override
+        public Void call(BlockWorkerClientService.Client client) throws TException {
+          client.sessionHeartbeat(mSessionId, null);
+          return null;
+        }
+      });
```

Comment:
good point Calvin, I addressed this by allowing to pass a RetryPolicy into the session heartbeat method